### PR TITLE
Add code coverage target to CMake

### PR DESCRIPTION
**Pull request by rbberger**
## Purpose

This adds the `-DENABLE_COVERAGE` option to the cmake build. Object files will be compiled with the `--coverage` flag, which produces gcov files during execution inside of the build directory.

That means, any testcase suite that runs the generated `lmp` binary will also generate output which tracks which lines of code have been executed.

The `gen_coverage_xml` target then generates a XML report file `coverage.xml` which can be imported into Jenkins Cobertura Plugin (https://wiki.jenkins.io/display/JENKINS/Cobertura+Plugin).

`gen_coverage_html` generates HTML files, starting with `coverage.html`, which showcase the code coverage.

### Example:
```bash
# build with coverage option
mkdir build
cd build/
cmake ../cmake -DENABLE_COVERAGE=on
make -j 12

# run melt example with lmp binary
cd ../examples/melt/
../../build/lmp -in in.melt

# generate coverage report
cd ../../build/
make gen_coverage_xml
make gen_coverage_html

# open in a browser
firefox coverage.html
```

## Author(s)
@rbberger

## Backward Compatibility
Yes.

## Implementation Notes
Requires gcovr to be installed. For support of GCC 8.1, make sure to get the latest version via pip.

```
pip install git+https://github.com/gcovr/gcovr.git
```

## Post Submission Checklist

_Please check the fields below as they are completed_
- [x] The feature or features in this pull request is complete
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] One or more example input decks are included
- [ ] The source code follows the LAMMPS formatting guidelines

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -49,6 +49,11 @@ if (${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
   set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -restrict")
 endif()
 
+option(ENABLE_COVERAGE "Enable code coverage" OFF)
+if(ENABLE_COVERAGE)
+  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage")
```
Comment (junghans) [inline cmake/CMakeLists.txt]: Doesn't it need to go in `CMAKE_EXE_LINKER_FLAGS` and `CMAKE_SHARED_LINKER_FLAGS` as well?

File: cmake/Modules/FindGCOVR.cmake [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#  GCOVR_BINARY       - path to gcovr executable
+#  GCOVR_FOUND        - True if gcovr found.
+#
+find_program(GCOVR_BINARY gcovr)
+
+include(FindPackageHandleStandardArgs)
+# handle the QUIETLY and REQUIRED arguments and set GCOVR_FOUND to TRUE
+# if all listed variables are TRUE
+
+find_package_handle_standard_args(GCOVR DEFAULT_MSG GCOVR_BINARY)
```
Comment (junghans) [inline cmake/Modules/FindGCOVR.cmake]: I did the same inline for `gzip` and `ffmpeg`, for such simple module that should be fine.

File: cmake/CMakeLists.txt (hunk 1 above)
Comment (rbberger) [reply cmake/CMakeLists.txt]: apparently not (output of `make VERBOSE=1`):
```
/usr/bin/c++   --coverage -fopenmp -O3 -DNDEBUG  -rdynamic CMakeFiles/lmp.dir/home/richard/GitHub/rbberger/lammps/src/main.cpp.o  -o lmp -Wl,-rpath,/usr/lib/openmpi: liblammps.a /usr/lib/openmpi/libmpi_cxx.so /usr/lib/openmpi/libmpi.so -ljpeg -lpng -lz -lm 
```

File: cmake/Modules/FindGCOVR.cmake (hunk 2 above)
Comment (rbberger) [reply cmake/Modules/FindGCOVR.cmake]: Ok, simplifying it.

**Pull request merged**